UnitTests/GitCommandsTests/Git/ExecutableExtensionsTests.cs

```diff
@@ -71,9 +71,15 @@ public void GetOutput_with_cache_miss()
         // Process argument upper bound is actually (short.MaxValue - 1)
         // 18: '"cmd.exe" /C echo '
         // 10: '"cmd.exe" '
-        [TestCase(short.MaxValue - 1 - 18, short.MaxValue - 1 - 18, 10)]
-        [TestCase(short.MaxValue - 1 - 19, 1, 10)]
-        public void RunBatchCommand_can_handle_max_length_arguments(int arg1Len, int arg2Len, int baseLength)
+        // 1: ' ' character
+        // cmd.exe max length is 8191, including NULL Unicode -> 8189
+        // https://support.microsoft.com/en-us/help/830473/command-prompt-cmd-exe-command-line-string-limitation
+        [TestCase(8189 - 1 - 18, 8189 - 1 - 18, 10, 8191, true)]
+        [TestCase(8189 - 1 - 19, 1, 10, 8191, true)]
+        [TestCase(8189 - 1 - 18, 8189 - 1 - 18, 10, 32767, true)]
+        [TestCase(8189 - 1 - 19, 1, 10, 32767, true)]
+        public void RunBatchCommand_can_handle_max_length_arguments(int arg1Len, int arg2Len, int appLength,
+            int maxLength, bool expectedResult)
```

Comment:
I liked the attempt to test the maximum length of a command line.
You could switch to `sh.exe -c "echo abc..."`. This executable should exist in `AppSettings.GitBinDir`.

Reply:
When I run NUnit on my local machine, it resolves GitExtensions.Setting to `\AppData\Roaming\Microsoft Corporation\Microsoft.TestHost.x86\` instead of  `\AppData\Roaming\GitExtensions\GitExtensions`. How can I fix this?

Reply:
That's a pity. There must be other NUnit tests which access the Git executable. These must access the GE settings, too. I haven't digged into this area yet.

Reply:
I've searched all the codes, they've only access `AppSettings` for setting bits that are not dependent on `Application.UserAppDataPath`, such as `AppSettings.AvatarImageCachePath`.

I've tried to manually create `SettingContainers` as below, but somehow it can't load settings in the correct folder.
```CSharp
var settingPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "GitExtensions\\GitExtensions");
var settingContainer = new RepoDistSettings(null, GitExtSettingsCache.Create(settingPath), SettingLevel.Unknown);
// gitbindir is always null
var appPath = Path.Combine(settingContainer.GetString("gitbindir", ""), "sh.exe");
```

Reply:
My bad, I've got it working. Correct `"GitExtensions\\GitExtensions"` to `"GitExtensions\\GitExtensions\\GitExtensions.settings"`. Works fine now.

Reply:
@mstv `sh.exe -c echo "abcd..."` will show below error at `32752` characters, `15` char short vs docs.
> 
> error launching git: The filename or extension is too long.
> 
Does that mean internally, sh.exe is launching `git`?

On the other hand, `git.exe reset -- "abcd..."` will correctly work until `35765` character (tested by create a batch script and execute it) with `%ERRORLEVEL% = 0`. However, when run in unit test, this command always returns non-zero exit code regardless of command length.

So I'm stuck at unit testing the exact threshold at the moment. `sh.exe` won't reach it, `git.exe` always return non-zero exit code.

Reply:
I was able to resolve the above issue, by setting `workingDir` of `git.exe` executable to local git repository using current Assembly location. Commited changes and passed test case with exact threshold of `32766` characters